e2e/components/menu/menu.ts

```diff
@@ -23,84 +23,60 @@
+
+  uploadFilesInput = this.byId('app-upload-files', browser);
+  submenus = this.byCss('app-context-menu-item .mat-menu-item', browser);
+
+  uploadFileAction = this.byId('app.create.uploadFile');
+  uploadFolderAction = this.byId('app.create.uploadFolder');
+  createFolderAction = this.byId('app.create.folder');
+  createLibraryAction = this.byId('app.create.library');
+  createFileFromTemplateAction = this.byId('app.create.fileFromTemplate');
+  createFolderFromTemplateAction = this.byId('app.create.folderFromTemplate');
+
+  cancelEditingAction = this.byCss(`.mat-menu-item[title='Cancel Editing']`);
+  cancelJoinAction = this.byCssText('.mat-menu-item', 'Cancel Join');
+  copyAction = this.byCssText('.mat-menu-item', 'Copy');
+  deleteAction = this.byCssText('.mat-menu-item', 'Delete');
+  downloadAction = this.byCssText('.mat-menu-item', 'Download');
+  editFolderAction = this.byCss(`.mat-menu-item[id$='editFolder']`);
+  editOfflineAction = this.byCss(`.mat-menu-item[title='Edit Offline']`);
+  favoriteAction = this.byCss(`.mat-menu-item[id$='favorite.add']`);
+  removeFavoriteAction = this.byCss(`.mat-menu-item[id$='favorite.remove']`);
+  toggleFavoriteAction = this.byCssText('.mat-menu-item', 'Favorite');
+  toggleRemoveFavoriteAction = this.byCssText('.mat-menu-item', 'Remove Favorite');
+  joinAction = this.byCssText('.mat-menu-item', 'Join');
+  leaveAction = this.byCssText('.mat-menu-item', 'Leave');
+  managePermissionsAction = this.byCssText('.mat-menu-item', 'Permissions');
+  manageVersionsAction = this.byCssText('.mat-menu-item', 'Manage Versions');
+  uploadNewVersionAction = this.byCssText('.mat-menu-item', 'Upload New Version');
+  moveAction = this.byCssText('.mat-menu-item', 'Move');
+  permanentDeleteAction = this.byCssText('.mat-menu-item', 'Permanently Delete');
+  restoreAction = this.byCssText('.mat-menu-item', 'Restore');
+  shareAction = this.byCssText('.mat-menu-item', 'Share');
+  shareEditAction = this.byCssText('.mat-menu-item', 'Shared Link Settings');
+  viewAction = this.byCssText('.mat-menu-item', 'View');
+  viewDetailsAction = this.byCssText('.mat-menu-item', 'View Details');
 
   constructor(ancestor?: string) {
-    super(Menu.selectors.root, ancestor);
+    super('.mat-menu-panel', ancestor);
   }
 
   async waitForMenuToOpen(): Promise<void> {
-    await browser.wait(EC.presenceOf(browser.element(by.css('.cdk-overlay-container .mat-menu-panel'))), BROWSER_WAIT_TIMEOUT);
-    await browser.wait(EC.visibilityOf(this.items.get(0)), BROWSER_WAIT_TIMEOUT);
+    await waitForPresence(browser.element(by.css('.cdk-overlay-container .mat-menu-panel')));
+    await waitForVisibility(this.items.get(0));
   }
 
   async waitForMenuToClose(): Promise<void> {
-    await browser.wait(EC.not(EC.presenceOf(browser.element(by.css('.cdk-overlay-container .mat-menu-panel')))), BROWSER_WAIT_TIMEOUT);
+    await waitForStaleness(browser.element(by.css('.cdk-overlay-container .mat-menu-panel')));
```

Comment:
due to some protractor weirdness, when dealing with elements that appear and then disappear, EC.not(EC.presenceOf..) is NOT equivalent with EC.stalenessOf 
so I suggest to not wait for staleness in this case, and to use EC.not(EC.presenceOf(..) , cause otherwise this will result in test flakiness.

Reply:
See here: https://github.com/angular/protractor/blob/5d8da049815ffc104e6eca71d2ecaec51b00f520/lib/expectedConditions.ts#L369